app/src/main/kotlin/com/hippo/ehviewer/gallery/PageLoader.kt

```diff
@@ -155,5 +141,51 @@ abstract class PageLoader(val gid: Long, var startPage: Int) : AutoCloseable {
         FileUtils.sanitizeFilename("$title - ${index + 1}.${it.lowercase()}")
     }
 
+    fun request(index: Int) {
+        val prefetchRange = if (index >= prevIndex) {
+            index + 1..(index + prefetchPageCount).coerceAtMost(size - 1)
+        } else {
+            index - 1 downTo (index - prefetchPageCount).coerceAtLeast(0)
+        }
+        // Prefetch to disk
+        val pagesAbsent = prefetchRange.filter {
+            when (pages[it].status) {
+                PageStatus.Queued, is PageStatus.Error -> true
+                else -> false
+            }
+        }
+        prevIndex = index
+        val image = lock.read { cache[index] }
+        if (image != null) {
+            notifyPageSucceed(index, image, false)
+        } else {
+            pages[index].statusFlow.update { status ->
+                when (status) {
+                    is PageStatus.Error -> PageStatus.Queued
+                    else -> status
+                }
+            }
+            onRequest(index)
+        }
```

Comment:
```suggestion
        prevIndex = index
        val image = lock.read { cache[index] }
        if (image != null) {
            notifyPageSucceed(index, image, false)
        } else {
            notifyPageWait(index)
            onRequest(index)
        }

        // Prefetch to disk
        val pagesAbsent = prefetchRange.filter {
            when (pages[it].status) {
                PageStatus.Queued, is PageStatus.Error -> true
                else -> false
            }
        }
```
So the comment matches the code.